src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx

```diff
@@ -147,18 +147,24 @@ export default class OrganizationDiscover extends React.Component {
 
     clearIndicators();
 
-    resultManager.fetchAll().then(data => {
-      const shouldRedirect = !this.props.params.savedQueryId;
-
-      if (shouldRedirect) {
-        browserHistory.push({
-          pathname: `/organizations/${organization.slug}/discover/`,
-          search: getQueryStringFromQuery(queryBuilder.getInternal()),
-        });
-      }
-
-      this.setState({data, isFetchingQuery: false});
-    });
+    resultManager
+      .fetchAll()
+      .then(data => {
+        const shouldRedirect = !this.props.params.savedQueryId;
+
+        if (shouldRedirect) {
+          browserHistory.push({
+            pathname: `/organizations/${organization.slug}/discover/`,
+            search: getQueryStringFromQuery(queryBuilder.getInternal()),
+          });
+        }
+
+        this.setState({data, isFetchingQuery: false});
+      })
+      .catch(err => {
```

Comment:
What's `err` here? Is it going to be a response from the API?

Reply:
Yes, either that or it's failed one of the client side validations